sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -201,6 +199,12 @@ namespace Azure { namespace Core {
           = ContextSharedState::ToMsecSinceEpoch((time_point::min)());
     }
 
+    /**
+     * @brief Gets the time point when this context will be cancelled.
+     * @return The time point when this context will be cancelled.
+     */
+    time_point CancelWhen() const;
```

Comment:
`CancelAt()` ?

Reply:
`CancelAt()` is fine by me. But I think it should be done in a separate PR.

Reply:
I'm gonna open an issue to track that.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/2032

Reply:
What about unit test and changelog update? We need either a fix here, or an issue marked with `blocking-release`, so we do mention it in this release changelog. First is better with me.

Reply:
we need the changelog for Azure Core for adding the method, yes.

Reply:
@antkmsft @vhvb1989 

Since CL and UT involve a lot of core sdk work, which I'm not familiar with. I think it'd be more appropriate if you guys can work on it. I updated the issue https://github.com/Azure/azure-sdk-for-cpp/issues/2032

I'll mark this PR as draft for now.

Reply:
@Jinming-Hu, it could be #2033, if my team is ok with all the changes I made there.